common/core/src/commonMain/kotlin/com/trafi/analytics/Analytics.kt

```diff
@@ -0,0 +1,31 @@
+package com.trafi.analytics
+
+import kotlin.native.concurrent.ThreadLocal
+
+interface AnalyticsDelegate {
+    fun dispatchEvent(event: Analytics.Event)
+
+    val screenName: String?
+    val modalName: String?
+}
+
+@ThreadLocal
+object Analytics {
```

Comment:
It would be great to have all these properties and functions static. In Swift we consume events through a static method: `Analytics.consume(event)`. Currently the generated interface requires to instantiate the `Analytics` type and call methods from there.

Reply:
Please experiment to see what Kotlin code gives you the result you expect. Ping me if you'd like to brainstorm ideas together.